libs/features/quest/src/quest-preview/quest-preview.model.ts

```diff
@@ -11,6 +11,8 @@ export interface DifficultyLevel {
   grey?: number;
 }
 
+export type QuestFactionRewardKey = keyof typeof QUEST_FACTION_REWARD;
+export type QuestFactionRewardValue = (typeof QUEST_FACTION_REWARD)[QuestFactionRewardKey];
```

Comment:
maybe something like? to avoid repeating `typeof QUEST_FACTION_REWARD`

```suggestion
export type QuestFactionRewardType = typeof QUEST_FACTION_REWARD;
export type QuestFactionRewardKey = keyof QuestFactionRewardType;
export type QuestFactionRewardValue = (QuestFactionRewardType)[QuestFactionRewardKey];
```